source/win-spout.cpp

```diff
@@ -8,31 +8,61 @@
  */
 
 #include <obs-module.h>
+#include <obs-frontend-api.h>
 #include <sys/stat.h>
+#include <QAction>
+#include <QMainWindow>
 
 #include "win-spout.h"
+#include "ui/win-spout-output-settings.h"
+#include "win-spout-config.h"
 
 OBS_DECLARE_MODULE()
 OBS_MODULE_AUTHOR("Off World Live")
 OBS_MODULE_USE_DEFAULT_LOCALE("win-spout", "en-US")
 
-SPOUTHANDLE spoutptr = NULL;
-
 extern struct obs_source_info create_spout_source_info();
 struct obs_source_info spout_source_info;
 
+extern struct obs_output_info create_spout_output_info();
+struct obs_output_info spout_output_info;
+
+win_spout_output_settings* spout_output_settings;
+obs_output_t* win_spout_out;
+
 bool obs_module_load(void)
 {
-	spoutptr = GetSpout();
 
-	if (!spoutptr) {
-		blog(LOG_ERROR, "can't get the spout pointer!");
+	spout_source_info = create_spout_source_info();
+	obs_register_source(&spout_source_info);
+
+	QMainWindow* main_window = (QMainWindow*)obs_frontend_get_main_window();
+
+	if (!main_window) {
+		blog(LOG_ERROR, "Can't get main window!");
 		return false;
 	}
-	blog(LOG_INFO, "Spoutlibrary initialized successfully");
 
-	spout_source_info = create_spout_source_info();
-	obs_register_source(&spout_source_info);
+	win_spout_config* config = win_spout_config::get();
+	config->load();
+
+	spout_output_info = create_spout_output_info();
+	obs_register_output(&spout_output_info);
+
+	obs_data_t* settings = obs_data_create();
+	win_spout_out = obs_output_create("spout_output", "OBS Spout Output", settings, NULL);
+	obs_data_release(settings);
+
+	QAction* menu_action =
+		(QAction*)obs_frontend_add_tools_menu_qaction(
+		obs_module_text("toolslabel"));
+
+	obs_frontend_push_ui_translation(obs_module_get_string);
+	spout_output_settings = new win_spout_output_settings(main_window);
```

Comment:
Does this not need to be deleted on unload?

Reply:
Good catch, we should add that in!

Reply:
appears that calling `delete` on this crashes OBS, but without it OBS closes correctly even opening multiple versions of obs and closing, so assuming  that this is not a memory leak we need to worry about